staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -330,6 +330,14 @@ func addAuditAnnotationLocked(ac *AuditContext, key, value string) {
 	ae.Annotations[key] = value
 }
 
+// AuditEventFrom returns a deep copy of the audit event struct on the ctx.
+func AuditEventFrom(ctx context.Context) *auditinternal.Event {
+	if ac := AuditContextFrom(ctx); ac.Enabled() {
+		return ac.event.DeepCopy()
```

Comment:
This is an unprotected read of event, which, if this function was used, would reintroduce the possibility of data races. Any access to the event needs to be done under lock, or using the ac.visitEvent helper

Reply:
I'm sorry I missed that. Thanks for pointing that out!

Fixed and force pushed.